.changeset/many-garlics-lick.md

```diff
@@ -0,0 +1,14 @@
+---
+'astro': major
+---
+
+Removes internal JSX handling as they are now handled by `@astrojs/mdx` directly. The below exports are also now removed:
+
+- `astro/jsx/babel.js`
+- `astro/jsx/component.js`
+- `astro/jsx/index.js`
+- `astro/jsx/renderer.js`
+- `astro/jsx/server.js`
+- `astro/jsx/transform-options.js`
+
+Make sure to upgrade `@astrojs/mdx` to latest so that it doesn't rely on these entrypoints.
```

Comment:
```suggestion
If your project includes `.mdx` files, you must upgrade `@astrojs/mdx` to the latest version so that it doesn't rely on these entrypoints to handle your JSX.
```